docs/checks/app_block_valid_tags.md

```diff
@@ -0,0 +1,41 @@
+# Reject Invalid Tags for Theme App Extension Blocks (`AppBlockValidTags`)
+
+This rule exists to prevent invalid tags from being used in theme app extension blocks which will be rejected when the theme app extension is promoted.
```

Comment:
>This rule exists to prevent invalid tags
- `{% javascript %}`
 - `{% stylesheet %}`...etc are actually valid tags however they are not allowed to be used in the context of AppBlocks, wondering if the wording can be improved? 


>theme app extension is promoted.

I would keep this platform specific operation `promotion` out of theme_check dictionary.